src/ServiceControl.Transports/TransportManifest.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace ServiceControl.Transports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    public class TransportManifest
+    {
+        public string Version { get; set; }
+        public List<TransportManifestDefinition> Definitions { get; set; }
+    }
+
+    public class TransportManifestDefinition
+    {
+        public string Name { get; set; }
+        public string DisplayName { get; set; }
+        public string TypeName { get; set; }
+        public List<string> Aliases { get; set; }
+    }
+
+    public static class TransportManifestLibrary
+    {
+        public static List<TransportManifest> TransportManifests { get; set; }
+
+        static bool initialized;
+        static void Initialize()
+        {
+            if (TransportManifests == null)
+            {
+                TransportManifests = new List<TransportManifest>();
+            }
+
+            if (!initialized)
+            {
+                initialized = true;
+                var assemblyLocation = GetEntryOrExecutingAssemblyDirectory();
+                Directory.EnumerateFiles(assemblyLocation, "transport.manifest", SearchOption.AllDirectories).ToList().ForEach(manifest =>
```

Comment:
We might be able to use `TransportManifests.AddRange(..)` instead of doing a `.ToList(..)`.ForEach(..)`

Reply:
changed